tests/integration/targets/systemd_creds_encrypt/tasks/main.yaml

```diff
@@ -53,3 +53,59 @@
           - '"SetCredentialEncrypted=web: " in pretty_encrypted_secret.value'
         fail_msg: "SetCredentialEncrypted is not in the output"
         success_msg: "SetCredentialEncrypted is in the output"
+
+    - name: Ensure the dest file doesn't already exist
+      ansible.builtin.file:
+        path: /tmp/systemd_cred_encrypt_test_secret.cred
+        state: absent
+
+    - name: Encrypt secret into the dest file
+      community.general.systemd_creds_encrypt:
+        secret: token
+        dest: /tmp/systemd_cred_encrypt_test_secret.cred
+      register: encrypted_dest
+
+    - name: Assert that the task changes when no secret was present
+      ansible.builtin.assert:
+        that:
+          - "encrypted_dest.changed"
+        fail_msg: "The task didn't change, but should have"
+        success_msg: "The task changed, as it should have"
+
+    - name: Check if the dest file exists
+      stat:
+        path: /tmp/systemd_cred_encrypt_test_secret.cred
+      register: dest_file
+
+    - name: Assert that the dest file exists
+      ansible.builtin.assert:
+        that:
+          - "dest_file.stat.exists"
+        fail_msg: "The dest file doesn't exist"
+        success_msg: "The dest file does exist"
+
+    - name: Encrypt the same secret into the same dest file as before
+      community.general.systemd_creds_encrypt:
+        secret: token
+        dest: /tmp/systemd_cred_encrypt_test_secret.cred
+      register: encrypted_dest_again
+
+    - name: Assert that the task doesn't change with the same secret
+      ansible.builtin.assert:
+        that:
+          - "not encrypted_dest_again.changed"
+        fail_msg: "The task changed, but shouldn't have"
+        success_msg: "The task didn't change, as it should have"
+
+    - name: Encrypt a different secret into the same dest file as before
+      community.general.systemd_creds_encrypt:
+        secret: another token
+        dest: /tmp/systemd_cred_encrypt_test_secret.cred
+      register: encrypted_dest_again
+
+    - name: Assert that the task changes with a different secret
+      ansible.builtin.assert:
+        that:
+          - "encrypted_dest_again.changed"
+        fail_msg: "The task didn't change, but should have"
+        success_msg: "The task changed, as it should have"
```

Comment:
Perhaps add an decryption test as well?

Reply:
or https://github.com/ansible-collections/community.general/pull/10549/files#diff-b0129a525eebd2b35f800b7d9ef6b272d5824976e0b9e42f5050892c87774542R144 is perhaps enough